core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/BlueprintUpgradeOptionValidator.java

```diff
@@ -18,17 +20,20 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(BlueprintUpgradeOptionValidator.class);
 
+    @Inject
+    private CustomTemplateUpgradeValidator customTemplateUpgradeValidator;
+
     boolean isValidBlueprint(Blueprint blueprint, boolean lockComponents) {
         LOGGER.debug("Validating blueprint upgrade option. Name: {}, type: {}, upgradeOption: {}, lockComponents: {}", blueprint.getName(),
                 blueprint.getStatus(), blueprint.getBlueprintUpgradeOption(), lockComponents);
-        return isCustomBlueprint(blueprint) || isEnabled(lockComponents, blueprint);
+        return isDefaultBlueprint(blueprint) ? isEnabledForDefaultBlueprint(lockComponents, blueprint) : isEnabledForCustomBlueprint(blueprint);
```

Comment:
this is a follow-up question, but don't we want to indicate why the blueprint is not valid for an upgrade to the end-user? cc: @keyki